server/routes/notification.js

```diff
@@ -0,0 +1,224 @@
+const authMiddleware = require("../middlewares/authMiddleware");
+const ListProductModel = require("../models/listProductModel");
+const NotificationModel = require("../models/notificationModel");
+const { sequelize } = require("../models/productModel");
+const ProductModel = require("../models/productModel");
+const UserListModel = require("../models/userListModel");
+const { PRICE } = require("../utils/enums");
+const router = require("express").Router();
+
+//Make the given notification id read
+router.put("/read/:id", authMiddleware, async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { id: user_id } = req.user;
+
+    await NotificationModel.update(
+      { isRead: true },
+      {
+        where: {
+          id,
+          user_id,
+        },
+      }
+    );
+
+    //update returns only a 1 or a 0 so we create another query to show the client
+    const newNotification = await NotificationModel.findOne({
+      where: {
+        id,
+        user_id,
+      },
+    });
```

Comment:
We should only have to query once. instead of querying the database to find the notification to update it, then querying the database again to return it.

Reply:
Will this process of using a model.save() be better or does it in effect does the same number of database interactions?

const notification = NotificationModel.findOne()

notification.isRead = true

const updatedNotification = await notification.save() 

res.send(updatedNotifications)

Reply:
I _think_ you can add `returning: true` as an option when updating and it will return that new record. Take a look at https://sequelize.org/master/class/lib/model.js~Model.html#static-method-update